### DISCUSSION TITLE
Possibility to show "Next" with day if there is more than a week

### DISCUSSION THREAD

**Discussion by hassanasad** [js-fns/date-fns#3253] [Q&A]
I love this date-fns library. Its much robust to use.

Lately i have been tasked to show just the "day" from a date like "Tuesday". However if the date refers to a next week's Tuesday for instance, it should say "Next Tuesday" instead of just "Tuesday". I was wondering if there is a utility method in date-fns to be able to detect this?

In the documentation there are methods `isSameWeek` and `isThisWeek` but i think they do something different.

**Comment by fturmel** [comment]
Try using `differenceInCalendarWeeks` for this: https://date-fns.org/docs/differenceInCalendarWeeks

The previous week should return `-1`, same week `0` and next week `1`. If it's another value, it means the interval is greater than one week, so you can determine how to format these according to your requirements.

Good luck!

**Comment by hassanasad** [comment]
Thanks @fturmel - I will give it a try :)
